File: OP_PROJECT_JAVA/src/game/Plant.java
package game;

import java.util.Random;

public class Plant extends Organism {
    private static final int PLANT_INIT = 0;

    public Plant(String type, int strength, Position position, World world) {
        super(type, strength, PLANT_INIT, position, world);
    }

    @Override
    public void action() {
        int pr = 15;
        Random random = new Random();
        int probability = random.nextInt(100);

        if (probability < pr) {
            int direction = random.nextInt(4);
            Position newPosition = new Position(position.getY(), position.getX());

            settingDirection(direction, newPosition);

            if (world.isPositionWithinBounds(newPosition)
                    && world.getOrganismAtPosition(newPosition) == null) {
                Plant newPlant = new Plant(type, strength, newPosition, world);
                world.getMap().get(newPosition.getY()).set(newPosition.getX(), newPlant);
                world.getOrganisms().add(newPlant);
            }
        }
    }

    @Override
    public OrganismGroup getGroup() {
        return OrganismGroup.PLANT;
    }
}

File: OP_PROJECT_PYTHON/src/game/Plant.py
from random import randint
from Organism import Organism
from Position import Position


class Plant(Organism):
    PLANT_INIT = 0

    def __init__(self, organism_type, strength, position, world):
        super().__init__(organism_type, strength, Plant.PLANT_INIT, position, world)

    def action(self):
        pr = 15
        probability = randint(0, 99)

        if probability < pr:
            direction = randint(0, 3)
            new_position = Position(self.position.get_y(), self.position.get_x())
            self.setting_direction(direction, new_position)

            if self.world.is_position_within_bounds(new_position) \
                    and self.world.get_organism_at_position(new_position) is None:
                self.world.born_organism(self.type, new_position)

    def get_group(self):
        return "PLANT"
